CONTRIBUTING.md

```diff
@@ -41,90 +41,95 @@ Unit tests are nice for atomic components, but since jdbi is a complex ecosystem
 
 Since our tests essentially describe and verify jdbi's behavior, changing their specifics where it isn't inherently necessary is considered a red flag.
 
+The use of mocks and mocking framework is generally discouraged. There are a number of existing tests that use Mockito and they are a pain to maintain.
```

Comment:
It's not obvious why tests with mocking are a pain to maintain?

Reply:
the collide e.g. with the leak checker. I am not against using mocks if they are done right (and most of the time, you end up using stubs or dummies, not mocks). What is abhorred are the pseudo-"convenient", reflection-driven, most of the time "record and regurgitate" frameworks.  

There are the inevitable links:
* https://martinfowler.com/bliki/TestDouble.html
* https://blog.cleancoder.com/uncle-bob/2014/05/14/TheLittleMocker.html

After every larger rewrite, some tests break even if functionality did not change. And most of the time, these are the few mockito tests that we have because they do some naive "record this, then this, then this, then check that these methods were called n times" checking and when rewriting code, these patterns change. So you end up testing the mock invocation, not the functionality. And that is a waste of time.